nibabel/cifti2/cifti2.py

```diff
@@ -0,0 +1,1115 @@
+
+    https://www.nitrc.org/forum/attachment.php?attachid=333&group_id=454&forum_id=1955
+
+'''
+from __future__ import division, print_function, absolute_import
+import re
+import collections
+
+import numpy as np
+
+from .. import xmlutils as xml
+from ..filebasedimages import FileBasedHeader, FileBasedImage
+from ..nifti2 import Nifti2Image
+
+
+class CIFTI2HeaderError(Exception):
+    """ Error in CIFTI2 header
+    """
+
+
+CIFTI_MAP_TYPES = ('CIFTI_INDEX_TYPE_BRAIN_MODELS',
+                   'CIFTI_INDEX_TYPE_PARCELS',
+                   'CIFTI_INDEX_TYPE_SERIES',
+                   'CIFTI_INDEX_TYPE_SCALARS',
+                   'CIFTI_INDEX_TYPE_LABELS')
+
+CIFTI_MODEL_TYPES = ('CIFTI_MODEL_TYPE_SURFACE',
+                     'CIFTI_MODEL_TYPE_VOXELS')
+
+CIFTI_SERIESUNIT_TYPES = ('SECOND',
+                          'HERTZ',
+                          'METER',
+                          'RADIAN')
+
+CIFTI_BrainStructures = ('CIFTI_STRUCTURE_ACCUMBENS_LEFT',
+                         'CIFTI_STRUCTURE_ACCUMBENS_RIGHT',
+                         'CIFTI_STRUCTURE_ALL_WHITE_MATTER',
+                         'CIFTI_STRUCTURE_ALL_GREY_MATTER',
+                         'CIFTI_STRUCTURE_AMYGDALA_LEFT',
+                         'CIFTI_STRUCTURE_AMYGDALA_RIGHT',
+                         'CIFTI_STRUCTURE_BRAIN_STEM',
+                         'CIFTI_STRUCTURE_CAUDATE_LEFT',
+                         'CIFTI_STRUCTURE_CAUDATE_RIGHT',
+                         'CIFTI_STRUCTURE_CEREBELLAR_WHITE_MATTER_LEFT',
+                         'CIFTI_STRUCTURE_CEREBELLAR_WHITE_MATTER_RIGHT',
+                         'CIFTI_STRUCTURE_CEREBELLUM',
+                         'CIFTI_STRUCTURE_CEREBELLUM_LEFT',
+                         'CIFTI_STRUCTURE_CEREBELLUM_RIGHT',
+                         'CIFTI_STRUCTURE_CEREBRAL_WHITE_MATTER_LEFT',
+                         'CIFTI_STRUCTURE_CEREBRAL_WHITE_MATTER_RIGHT',
+                         'CIFTI_STRUCTURE_CORTEX',
+                         'CIFTI_STRUCTURE_CORTEX_LEFT',
+                         'CIFTI_STRUCTURE_CORTEX_RIGHT',
+                         'CIFTI_STRUCTURE_DIENCEPHALON_VENTRAL_LEFT',
+                         'CIFTI_STRUCTURE_DIENCEPHALON_VENTRAL_RIGHT',
+                         'CIFTI_STRUCTURE_HIPPOCAMPUS_LEFT',
+                         'CIFTI_STRUCTURE_HIPPOCAMPUS_RIGHT',
+                         'CIFTI_STRUCTURE_OTHER',
+                         'CIFTI_STRUCTURE_OTHER_GREY_MATTER',
+                         'CIFTI_STRUCTURE_OTHER_WHITE_MATTER',
+                         'CIFTI_STRUCTURE_PALLIDUM_LEFT',
+                         'CIFTI_STRUCTURE_PALLIDUM_RIGHT',
+                         'CIFTI_STRUCTURE_PUTAMEN_LEFT',
+                         'CIFTI_STRUCTURE_PUTAMEN_RIGHT',
+                         'CIFTI_STRUCTURE_THALAMUS_LEFT',
+                         'CIFTI_STRUCTURE_THALAMUS_RIGHT')
+
+
+def _value_if_klass(val, klass, check_isinstance_or_none=True):
```

Comment:
Sorry, browser confounded me - here's what I meant:

I think the second argument is always True for the calls in this module, so how about:

```
def _check_instance(val, klass):
    """ Check `val` is instance of `klass` or None, and return
    
     Raise ValueError if check fails
     """
     if val is None or isinstance(val, klass):
        return val
     raise ValueError('Input must be None or class ' + klass.__name__)
```